### DISCUSSION TITLE
Select a field with AsyncNotifier

### DISCUSSION THREAD

**Discussion by eyNmXGfPTS23** [rrousselGit/riverpod#2953] [Q&A]
I have a problem and it seems that I am the only one experiencing this expect [this thread](https://stackoverflow.com/questions/76877938/riverpod-equivalent-of-select-to-an-asyncvalue-object-from-an-asyncnotifierp) on StackOverflow.

My problem is that I am using an AsyncNotifier since when I enter the page, I need to get data from an API. The data type is a custom Model with multiple fields inside because the page is a form. I want to have only one notifier for the whole page, getting the data from the API **and** updating the form with my notifier.

Naturally, I wanted to use the select since every field are seperated in their own widget. But when I update any field like this :
`state = AsyncValue<MyModel>.data(_state.copyWith(name: value));`
it rebuild all the fields where I have a `.watch`. I think this is because we are changing the value of the async so it consider that the whole page need to be rebuild. 
How can I achieve this ?

In my widget I am watching like this:
```
final bool value = ref.watch(
      menuCategoryPageNotifierProvider(
        shopId: shopId,
        categoryId: categoryId,
      ).select((AsyncValue<CategoryEntity> provider) => provider.value!.available
      ),);
```

My notifier look like this :

```
@riverpod
class MenuCategoryPage extends _$MenuCategoryPage {
  ShopDetailsRepository get _shopDetailsRepo => getIt<ShopDetailsRepository>();

  CategoryEntity get _state => state.value!;

  bool _availability = true;

  @override
  FutureOr<CategoryEntity> build({required String shopId}) async {
    // I get something from the API
  }

  void updateAvailability(bool value) async {
    state = AsyncValue<CategoryEntity>.data(_state.copyWith(available: value));
  }
}
```

**Comment by rrousselGit** [comment]
There's no issue in the update + select combination you described. That should filter updates as intended.

If that doesn't work out for you, the issue is elsewhere

**Reply by eyNmXGfPTS23** [reply]
Hi, sorry for the delay,

After investigating, indeed the problem were elsewhere.
I notice that it was the function `.when` that was updating and since this widget is on top of my widget tree, it reload all the widget below. So, naturally, my question is there a way to not trigger the `.when` when I use the `updateAvailability` function in my provider. I understand that technically the `.when` trigger since we are giving a new async value but I would like to bypass that.

```
final AsyncValue<CategoryEntity> provider = ref.watch(
      MenuCategoryPageNotifierProvider(shopId: shopId),
    );

    return provider.when(
      data: (CategoryEntity category) => _Content(categoryId: categoryId),
      error: (Object error, _) => ErrorPageWidget(message: t.shopDetails.menuCategory.error.get, onRetry: () {}),
      loading: () => const _PageShimmer(),
    );
```

**Reply by eyNmXGfPTS23** [reply]
Hi, did you have time to look into the update about my problem ? @rrousselGit
